cartridge/lua-api/failover.lua

```diff
@@ -170,27 +171,30 @@ local function promote(replicaset_leaders, opts)
         return nil, PromoteLeaderError:new('There is no active coordinator')
     end
 
-    local _, err = rpc.call(
+    local ok, err = rpc.call(
             'failover-coordinator',
             'appoint_leaders',
             {replicaset_leaders},
             { uri = coordinator.uri }
     )
 
-    if err ~= nil then
+    if ok == nil then
         return nil, err
     end
 
-    if opts == nil or opts.force_inconsistency == false then
-        return true
+    if opts ~= nil and opts.force_inconsistency == true then
+        local ok, err = failover.force_inconsistency(replicaset_leaders)
+        if ok == nil then
+            return nil, PromoteLeaderError:new(
+                "Promotion succeeded, but inconsistency wasn't forced: %s",
+                errors.is_error_object(err) and err.err or err
```

Comment:
Why isn't it simply `tostring(err)`. Seems it should work in the same way for simple strings and errors from "errors" module

Reply:
tostring(err) includes stack traceback. Since I create new error object, i don't want it to become a part of new error message.